### PR TITLE
Relpacing GWClient with TCGClient

### PR DESCRIPTION
**Pull request by rhyas**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -190,6 +191,9 @@ type ConnectorDTO struct {
 // GWConnection defines Groundwork Connection configuration
 type GWConnection clients.GWConnection
 
+// GWConnection defines Groundwork Connection configuration
```
Comment (ymkins) [inline config/config.go]: should be TCGConnection 

https://github.com/golang/go/wiki/CodeReviewComments#doc-comments

File: config/config.go [hunk 2]
```diff
@@ -228,6 +232,19 @@ func (con *GWConnection) UnmarshalYAML(unmarshal func(interface{}) error) error
 	return nil
 }
 
+// Decode implements envconfig.Decoder interface
+// merges incoming value with existed structure
+func (con *TCGConnection) Decode(value string) error {
+	var overrides GWConnection
```
Comment (ymkins) [inline config/config.go]: should be TCGConnection

File: sdk/clients/tcgClient.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package clients
+
+import (
+	"context"
+	"github.com/rs/zerolog/log"
```
Comment (ymkins) [inline sdk/clients/tcgClient.go]: `sdk` module should avoid 3rd party deps
"github.com/gwos/tcg/sdk/logper" should be used instead

File: sdk/go.mod [hunk 4]
```diff
@@ -1,3 +1,5 @@
 module github.com/gwos/tcg/sdk
 
-go 1.17
+go 1.18
+
+require github.com/rs/zerolog v1.26.1
```
Comment (ymkins) [inline sdk/go.mod]: `sdk` module should avoid 3rd party deps
there is "github.com/gwos/tcg/sdk/logper"

**Review by ymkins — CHANGES_REQUESTED** [review]
There is `sdk` sub-module provided to support integrations. It should not include 3rd party dependencies.
For example, it's used in [influxdata/telegraf](https://github.com/influxdata/telegraf/tree/master/plugins/outputs/groundwork).

For today, having sub-module in repo complicates development workflow.
Probably, we can simplify it with modern go-1.18 https://go.dev/doc/tutorial/workspaces

File: sdk/clients/tcgClient.go (hunk 3 above)
Comment (rhyas) [reply sdk/clients/tcgClient.go]: Moved the client out of the SDK for now.

File: sdk/go.mod (hunk 4 above)
Comment (rhyas) [reply sdk/go.mod]: Moved the client out of the SDK for now.

File: config/config.go (hunk 2 above)
Comment (rhyas) [reply config/config.go]: Fixed

File: config/config.go (hunk 1 above)
Comment (rhyas) [reply config/config.go]: Fixed.

**Review by ymkins — APPROVED** [review]

File: sdk/go.sum [hunk 5]
```diff
@@ -0,0 +1,32 @@
+github.com/coreos/go-systemd/v22 v22.3.2/go.mod h1:Y58oyj3AT4RCenI/lSvhwexgC+NSVTIJ3seZv2GcEnc=
```
Comment (ymkins) [inline sdk/go.sum]: it looks like this file (sdk/go.sum) was added by some tooling
and it can be safely removed

**Review by ymkins — CHANGES_REQUESTED** [review]
please remove sdk/go.sum file

File: sdk/go.sum (hunk 5 above)
Comment (rhyas) [reply sdk/go.sum]: Removed

**Pull request merged**